### ISSUE TITLE
Transform gradient

### ISSUE THREAD

**Issue by fsx950223** [keras-team/keras#14945]
Please go to TF Forum for help and support:

https://discuss.tensorflow.org/tag/keras

If you open a GitHub issue, here is our policy:

It must be a bug, a feature request, or a significant problem with the documentation (for small docs fixes please send a PR instead).
The form below must be filled out.

**Here's why we have that policy:**.

Keras developers respond to issues. We want to focus on work that benefits the whole community, e.g., fixing bugs and adding features. Support only helps individuals. GitHub also notifies thousands of people when issues are filed. We want them to see you communicating an interesting problem, rather than being redirected to Stack Overflow.

**System information**.
- Have I written custom code (as opposed to using a stock example script provided in Keras):
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04):
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below): 2.5.0
- Python version:
- Bazel version (if compiling from source):
- GPU model and memory:
- Exact command to reproduce:

You can collect some of this information using our environment capture script:

https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh

You can obtain the TensorFlow version with:
python -c "import tensorflow as tf; print(tf.version.GIT_VERSION, tf.version.VERSION)"

**Describe the problem**.

**Describe the current behavior**.
I found ```_transform_gradients``` handle intermediate gradients instead of final grads (https://github.com/keras-team/keras/blob/aaa022c622f583482e7502725272de201495fe52/keras/optimizer_v2/optimizer_v2.py#L656) and the behavior is not expected. 

**Describe the expected behavior**.
change https://github.com/keras-team/keras/blob/aaa022c622f583482e7502725272de201495fe52/keras/optimizer_v2/optimizer_v2.py#L689 to
```python
      grad = self._resource_apply_dense(grad, var, **apply_kwargs)
      grad, var = self._transform_gradients(grad, var)
      update_op = var.assign(grad)
```
 would get expected result.

**[Contributing](https://github.com/keras-team/keras/blob/master/CONTRIBUTING.md)**.

- Do you want to contribute a PR? (yes/no):
- If yes, please read [this page](https://github.com/keras-team/keras/blob/master/CONTRIBUTING.md) for instructions
- Briefly describe your candidate solution(if contributing):

**Standalone code to reproduce the issue**.

Provide a reproducible test case that is the bare minimum necessary to generate
the problem. If possible, please share a link to Colab/Jupyter/any notebook.

**Source code / logs**.

Related https://github.com/tensorflow/addons/pull/2525

**Comment by chenmoneygithub** [comment]
Hi, thanks for reporting the issue!

Not quite sure here, but by 'final gradients', I guess you are referring to the incremental value of variables?  _transform_gradients() is designed to be applied on the original gradients.
